azurerm/internal/services/keyvault/key_vault_certificate_issuer_resource.go

```diff
@@ -0,0 +1,274 @@
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_key_vault_certificate", *existing.ID)
+		}
+	}
+
+	parameter := keyvault.CertificateIssuerSetParameters{}
+	parameter.Provider = utils.String(d.Get("provider_name").(string))
+	parameter.OrganizationDetails = &keyvault.OrganizationDetails{ID: utils.String(d.Get("org_id").(string))}
+	parameter.Credentials = &keyvault.IssuerCredentials{AccountID: utils.String(d.Get("account_id").(string)), Password: utils.String(d.Get("password").(string))}
+	resp, err := client.SetCertificateIssuer(ctx, keyVaultBaseUri, name, parameter)
+	if err != nil {
+		return fmt.Errorf("Error setting Certificate Issuer %q (Key Vault %q): %s", name, keyVaultId, err)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmKeyVaultCertificateIssuerRead(d, meta)
+}
+
+func resourceArmKeyVaultCertificateIssuerUpdate(d *schema.ResourceData, meta interface{}) error {
+	vaultClient := meta.(*clients.Client).KeyVault.VaultsClient
+	client := meta.(*clients.Client).KeyVault.ManagementClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	keyVaultId := d.Get("key_vault_id").(string)
+
+	keyVaultBaseUri, err := azure.GetKeyVaultBaseUrlFromID(ctx, vaultClient, keyVaultId)
+	if err != nil {
+		return fmt.Errorf("Error looking up Certificate Issuer %q vault url from id %q: %+v", name, keyVaultId, err)
+	}
+
+	existing, err := client.GetCertificateIssuer(ctx, keyVaultBaseUri, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("Error checking for presence of existing Certificate Issuer %q (Key Vault %q): %s", name, keyVaultBaseUri, err)
+		} else {
+			return fmt.Errorf("KeyVault Certificate Issuer %q (KeyVault URI %q) does not exist", name, keyVaultBaseUri)
+		}
+	}
+
+	parameter := keyvault.CertificateIssuerSetParameters{}
+	parameter.Provider = utils.String(d.Get("provider_name").(string))
+	parameter.OrganizationDetails = &keyvault.OrganizationDetails{ID: utils.String(d.Get("org_id").(string))}
+	parameter.Credentials = &keyvault.IssuerCredentials{AccountID: utils.String(d.Get("account_id").(string)), Password: utils.String(d.Get("password").(string))}
+	resp, err := client.SetCertificateIssuer(ctx, keyVaultBaseUri, name, parameter)
+	if err != nil {
+		return fmt.Errorf("Error setting Certificate Issuer %q (Key Vault %q): %s", name, keyVaultId, err)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmKeyVaultCertificateIssuerRead(d, meta)
+}
+
+func resourceArmKeyVaultCertificateIssuerRead(d *schema.ResourceData, meta interface{}) error {
+	vaultClient := meta.(*clients.Client).KeyVault.VaultsClient
+	client := meta.(*clients.Client).KeyVault.ManagementClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	keyVaultId := d.Get("key_vault_id").(string)
```

Comment:
We should get the information to make the `Read` call from the ID in the state (using `d.Id()`), rather than the config, this allows us to understand the difference between the config and what Terraform last knew about.